### PR TITLE
Add tests for vlob_* cmds

### LINKED ISSUE #6629: Server test vlob_create

### LINKED ISSUE #6630: Server test vlob_poll_changes

### LINKED ISSUE #6631: Server test vlob_read_batch

### LINKED ISSUE #6632: Server test vlob_read_versions

### LINKED ISSUE #6633: Server test vlob_update

### PR DESCRIPTION
**Pull request by mmmarcos**
Also, fix bug in vlob create/update: the condition for `RequireGreaterTimestamp` did not trigger on equal timestamps (it was reversed).

closes #6629
closes #6630
closes #6631
closes #6632
closes #6633

### CODE REVIEW

File: server/parsec/components/memory/vlob.py [hunk 1]
```diff
@@ -120,8 +120,11 @@ async def create(
         if maybe_error is not None:
             return maybe_error
 
-        if timestamp < org.last_certificate_timestamp:
-            return RequireGreaterTimestamp(strictly_greater_than=org.last_certificate_timestamp)
+        assert org.last_certificate_or_vlob_timestamp is not None  # Orga bootstrapped
+        if org.last_certificate_or_vlob_timestamp >= timestamp:
+            return RequireGreaterTimestamp(
+                strictly_greater_than=org.last_certificate_or_vlob_timestamp
+            )
```
Comment (mmmarcos) [inline server/parsec/components/memory/vlob.py:127]: @touilleMan can you confirm is these changes are OK? (see the PR description)

File: server/tests/api_v4/authenticated/test_vlob_read_batch.py [hunk 2]
```diff
@@ -3,6 +3,12 @@
 import pytest
 
 from parsec._parsec import DateTime, DeviceID, OrganizationID, VlobID, authenticated_cmds
+
+# TODO: fix pyright error
+# - error: "VLOB_READ_REQUEST_ITEMS_LIMIT" is unknown import symbol (reportGeneralTypeIssues)
+# from parsec.components.vlob import VLOB_READ_REQUEST_ITEMS_LIMIT
+VLOB_READ_REQUEST_ITEMS_LIMIT = 1000
```
Comment (mmmarcos) [inline server/tests/api_v4/authenticated/test_vlob_read_batch.py:10]: I'm not sure how to fix this pyright error.

File: server/tests/api_v4/authenticated/test_vlob_create.py [hunk 3]
```diff
@@ -136,8 +138,8 @@ async def test_authenticated_vlob_create_bad_key_index(
         case "index_2_unknown":
             bad_key_index = 2
 
-        case unknown:
-            assert False, unknown
+        case _:
```
Comment (touilleMan) [inline server/tests/api_v4/authenticated/test_vlob_create.py:141]: the `assert False, unknown` is less readable than this `raise Exception(...)`, but is used really often in the test codebase (in every test parametrized with a `kind` parameter consisting of a string)

I won't fight against this change, but I'd rather keep simple the way of writing this check (writing test is already annoying enough ^^)

File: server/tests/api_v4/authenticated/test_vlob_create.py [hunk 4]
```diff
@@ -213,4 +214,183 @@ async def test_authenticated_vlob_create_vlob_already_exists(
...
+    backend: Backend,
+    timestamp_offset: int,
+) -> None:
+    dt1 = DateTime.now()
+    dt2 = dt1.add(seconds=10)
+    bad_timestamp = dt2.subtract(seconds=timestamp_offset)
+
+    # 1) Create a first vlob
+
+    author = coolorg.alice.device_id
```
Comment (touilleMan) [inline server/tests/api_v4/authenticated/test_vlob_create.py:295]: Why should we create a first vlob ? I would say we only need to add a new certificate in this test

File: server/tests/api_v4/authenticated/test_vlob_update.py [hunk 5]
```diff
@@ -249,4 +250,218 @@ async def test_authenticated_vlob_update_bad_vlob_version(
...
+            hash_algorithm=HashAlgorithm.SHA256,
+            encryption_algorithm=SecretKeyAlgorithm.XSALSA20_POLY1305,
+            key_index=2,
+            realm_id=realm_id,
+            key_canary=b"",
+        ).dump_and_sign(coolorg.alice.signing_key),
+    )
+    assert isinstance(outcome, RealmKeyRotationCertificate)
+
+    # 2) Create second vlob with same or previous timestamp
```
Comment (touilleMan) [inline server/tests/api_v4/authenticated/test_vlob_update.py:389]: the comment should be about updating the vlob

File: server/tests/api_v4/authenticated/test_vlob_update.py [hunk 6]
```diff
@@ -249,4 +250,218 @@ async def test_authenticated_vlob_update_bad_vlob_version(
...
+    backend: Backend,
+    timestamp_offset: int,
+) -> None:
+    dt1 = DateTime.now()
+    dt2 = dt1.add(seconds=10)
+    bad_timestamp = dt2.subtract(seconds=timestamp_offset)
+
+    # 1) Create a first vlob
+
+    author = coolorg.alice.device_id
```
Comment (touilleMan) [inline server/tests/api_v4/authenticated/test_vlob_update.py:348]: We can also avoid creating a new vlob and fetch one created for wksp1 (typically the workspace manifest's should already exist and be accessible with `coolorg.wksp1_id`)

File: server/tests/api_v4/authenticated/test_vlob_create.py (hunk 3 above)
Comment (mmmarcos) [reply server/tests/api_v4/authenticated/test_vlob_create.py:141]: I've followed this approach https://github.com/Scille/parsec-cloud/pull/6620#discussion_r1514623030
But now that you mentioned, it's probably better to stay consistent with the existing code base.

File: server/tests/api_v4/authenticated/test_vlob_update.py (hunk 5 above)
Comment (mmmarcos) [reply server/tests/api_v4/authenticated/test_vlob_update.py]: Done

File: server/tests/api_v4/authenticated/test_vlob_update.py (hunk 6 above)
Comment (mmmarcos) [reply server/tests/api_v4/authenticated/test_vlob_update.py:347]: Done

File: server/tests/api_v4/authenticated/test_vlob_create.py (hunk 3 above)
Comment (mmmarcos) [reply server/tests/api_v4/authenticated/test_vlob_create.py]: Done

**Review by touilleMan — APPROVED** [review]

**Pull request merged**